VocaDbWeb.Core/Controllers/ControllerBase.cs

```diff
@@ -0,0 +1,254 @@
+		}
+
+		protected ActionResult HttpStatusCodeResult(HttpStatusCode code, string message)
+		{
+			Response.StatusCode = (int)code;
+			// TODO: implement Response.StatusDescription = message;
+
+			return Content((int)code + ": " + message);
+		}
+
+		protected void ParseAdditionalPictures(IFormFile mainPic, IList<EntryPictureFileContract> pictures)
+		{
+			ParamIs.NotNull(() => mainPic);
+			ParamIs.NotNull(() => pictures);
+
+			var additionalPics = Enumerable.Range(0, Request.Form.Files.Count)
+				.Select(i => Request.Form.Files[i])
+				.Where(f => f != null && f.FileName != mainPic.FileName)
+				.ToArray();
+
+			var newPics = pictures.Where(p => p.Id == 0).ToArray();
+
+			for (int i = 0; i < additionalPics.Length; ++i)
+			{
+				if (i >= newPics.Length)
+					break;
+
+				var contract = ParsePicture(additionalPics[i], "Pictures", ImagePurpose.Additional);
+
+				if (contract != null)
+				{
+					newPics[i].OriginalFileName = contract.OriginalFileName;
+					newPics[i].UploadedFile = contract.UploadedFile;
+					newPics[i].Mime = contract.Mime;
+					newPics[i].ContentLength = contract.ContentLength;
+					newPics[i].Purpose = contract.Purpose;
+				}
+			}
+
+			CollectionHelper.RemoveAll(pictures, p => p.Id == 0 && p.UploadedFile == null);
+		}
+
+		protected EntryPictureFileContract ParsePicture(IFormFile pictureUpload, string fieldName, ImagePurpose purpose)
+		{
+			EntryPictureFileContract pictureData = null;
+
+			if (Request.Form.Files.Count > 0 && pictureUpload != null && pictureUpload.Length > 0)
+			{
+				if (pictureUpload.Length > ImageHelper.MaxImageSizeBytes)
+				{
+					ModelState.AddModelError(fieldName, "Picture file is too large.");
+					return null;
+				}
+
+				if (!ImageHelper.IsValidImageExtension(pictureUpload.FileName))
+				{
+					ModelState.AddModelError(fieldName, "Picture format is not valid.");
+					return null;
+				}
+
+				pictureData = new EntryPictureFileContract(pictureUpload.OpenReadStream(), pictureUpload.ContentType, (int)pictureUpload.Length, purpose);
+				pictureData.OriginalFileName = pictureUpload.FileName;
+			}
+
+			return pictureData;
+		}
+
+		protected ActionResult Picture(PictureContract pictureData, string title)
+		{
+			if (pictureData?.Bytes == null || string.IsNullOrEmpty(pictureData.Mime))
+				return File(HttpContext.RequestServices.GetRequiredService<IHttpContext>().ServerPathMapper.MapPath("~/Content/unknown.png"), "image/png");
+
+			var ext = ImageHelper.GetExtensionFromMime(pictureData.Mime);
+
+			if (!string.IsNullOrEmpty(ext))
+			{
+				//var encoded = Url.Encode(title);
+				// Note: there is no good way to encode content-disposition filename (see http://stackoverflow.com/a/216777)
+				Response.Headers.Add("content-disposition", $"inline;filename=\"{title}{ext}\"");
+			}
+
+			return File(pictureData.Bytes, pictureData.Mime);
+		}
+
+		protected ActionResult LowercaseJson(object obj) => Content(JsonHelpers.Serialize(obj), "application/json");

```

Comment:
Can we implement `LowercaseJson` like this? [The original implementation](https://github.com/VocaDB/vocadb/blob/2b78a2b42702e20dbc870ea58ad8c832c1cfcea9/VocaDbWeb/Controllers/ControllerBase.cs#L214) makes use of `JsonNetResult`.

Reply:
If all JSON is serialized as camelCase by default, then yes.